ovirt-engine-keycloak.spec.in

```diff
@@ -0,0 +1,113 @@
+%global __jar_repack 0
+
+%global product_name Keycloak SSO for oVirt Engine
+
+%global make_common_opts \\\
+	-j1 \\\
+	BUILD_VALIDATION=0 \\\
+	PACKAGE_NAME=%{name} \\\
+	RPM_VERSION=%{version} \\\
+	RPM_RELEASE=%{release} \\\
+	PREFIX=%{_prefix} \\\
+	SYSCONF_DIR=%{_sysconfdir} \\\
+	DATAROOT_DIR=%{_datadir} \\\
+	%{?EXTRA_BUILD_FLAGS:EXTRA_BUILD_FLAGS="%{EXTRA_BUILD_FLAGS}"}
+
+
+
+########################################################
+#  Keycloak overlay package
+########################################################
+Name:		ovirt-engine-keycloak
+Version:	@RPM_VERSION@
+Release:	@RPM_RELEASE@%{?dist}
+Summary:	%{product_name}
+Group:		Virtualization/Management
+License:	ASL 2.0
+URL:		http://keycloak.org
+BuildArch:	noarch
+Source:		%{name}-@RPM_VERSION@.tar.gz
+
+BuildRequires:	unzip
+
+Requires:	ovirt-engine-wildfly
+
+%description
+Keycloak SSO for oVirt Engine.
+
+########################################################
+#  Keycloak overlay setup package
+########################################################
+%package setup
+Summary:	%{product_name} setup
+Group:		Virtualization/Management
+
+BuildRequires:	python3
+BuildRequires:	python3-devel
+
+Requires:	ovirt-engine-setup-plugin-ovirt-engine-common >= 4.4.0
+Requires:	python%{python3_pkgversion}-ovirt-setup-lib
+Requires:	%{name} >= %{version}
+
+%description setup
+Keycloak SSO for oVirt Engine installation setup package.
+
+
+########################################################
+#  Package customizations
+########################################################
+%pre
+%ovirt_create_user_engine
+
+%prep
+%setup -cq
+
+%build
+make %{make_common_opts}
+
+%install
+rm -fr "%{buildroot}"
+make %{make_common_opts} install DESTDIR=%{buildroot}
+
+# Unzip downloaded keycloak overlay package
+mkdir -p %{buildroot}%{_datadir}
+unzip -d %{buildroot}%{_datadir}/%{name} @KEYCLOAK_OVERLAY_ZIP@
+
+# install Readme
+install -d -m 0755 "%{buildroot}%{_docdir}/%{name}"
+install -m 0644 "%{_builddir}/%{name}-%{version}/README.md" "%{buildroot}%{_docdir}/%{name}/README.md"
+
+# prepare sym links from ovirt-engine-wildfly to relevant ovirt-engine-keycloak artifacts
+# that is required because keycloak overlay is supposed to be extracted inside Wildfly/EAP location
+# and for ease of future management we do not want to mix them, symlinks here is an acceptable trade off
+mkdir -p %{buildroot}%{_datadir}/ovirt-engine-wildfly/modules/system/layers
```

Comment:
Perhaps:
1. Create a macro for %{_datadir}/ovirt-engine-wildfly
2. Create relative symlinks? Both absolute and relative have their pros and cons. Not sure about this case.

Not sure the comment above is clear enough. "is supposed" is what JBOSS expects, right? Not a human.

Also: I do not remember anymore what we do with the engine itself, but I think it's slightly different, no? I see stuff in /var/lib/ovirt-engine/jboss_runtime/tmp etc.

Reply:
ad 1. ack
ad 2. Actually I could not decide myself as well, just chose one of the two available options hoping to get a feedback. Unless/until nobody has objections I will keep them as they are.

The links are rather 'what Keycloak expects, rather than human'.  
jboss_runtime is a 'generated' sort-of-temporary location. It works in a way that ovirt-engine.xml.in is used as main Wildfly/EAP configuration and during a startup project specific artifacts (ear, war etc) & libraries are extracted into that runtime location.
ovirt-engine-wildfly package provides the 'static' server internals. 
Now because of those symlinks the whole flow works just like before. Alternatively I could have used  'Keycloak + Wildfly' distribution and Keycloak would have become serparate service (new Wildfly instance installation started as separate processa)... but that would be another story.